### DISCUSSION TITLE
DataTable @rowHover event?

### DISCUSSION THREAD

**Discussion by dominiksulak** [primefaces/community#340] [PrimeVue] [answered]
Hello

is there something like @rowClick event on DataTable component but for hover/mouseover event? I've found some old discussions about this feature but right now i cant access such a feature, nor i cant find it in official documentation. So I am wondering whether there is rowHover featuer or some similiar functionality. I need to emit some data on row hover.

thanks for answer

**Comment by tugcekucukoglu** [comment]
There is `rowHover` property for DataTable. https://primevue.org/datatable/#api.datatable.props.rowHover

**Comment by dominiksulak** [comment]
I know about this one, but I wasn't able to emit values onHover. According to link you sent: "When enabled, background of the rows change on hover." So I assume that this property is only for visual background changes. If I am not right and it is somehow possible to emit event (like @rowClick) please tell me how. 

Thank you!

**Reply by TeoDar** [reply, marked as answer]
Perhaps [this example](https://stackblitz.com/edit/2xcs8n?file=src%2FApp.vue) will help you understand how to call an event by hover on a row. In PrimeVue, as far as I understand, Pass Through is used for this.

**Reply by dominiksulak** [reply]
Yes this is exactly what I needed. I works perfectly, already implemented in my code. Thank you very much <3!

**Reply by chezhouse** [reply]
Your demo works fine, but for whatever reason, I'm getting a row index of 1 for all rows.  Not sure why yet, but I'll try and figure it out.

**Reply by haizk** [reply]
> Your demo works fine, but for whatever reason, I'm getting a row index of 1 for all rows. Not sure why yet, but I'll try and figure it out.

You can use `event.target.__vnode.props['data-p-index']` to get the row index, or `event.target.getAttribute('data-p-index')`.
Or you even can get the details with `event.target.__vnode.ctx.props.rowData`.

**Comment by Schnusiman** [comment]
Hi all,
I'm happy to find this solution.
Unfortunatly this is not working in a typescript enviroment or at least, I do not see the error:
I used the referenced example in a typescript example project and copied exactly the relevant files (*.css, productservice.ts, app.vue and main.ts), changed <script to <script lang="ts". To make it easier, I change the pt-parameter to:
`        :pt="{
        bodyRow: {
          onfocus: () => {console.log('focus');},
        },`
In the JS-project, I see, that the listener is added to the row, in the ts-project not: 
**JS-Project**
![js_listener](https://github.com/primefaces/community/assets/123939453/d08afbfb-2808-435f-b4fa-5e35e23a0493)
**TS-Project**
![ts_listener](https://github.com/primefaces/community/assets/123939453/e5184d1d-644b-42a8-8fb9-ba2f3b8b691c)

Is there a different syntax for to add a passthroug for typescript?

Thanks for any help here.

greetings

Here is [ a Demo on stackblitz](https://stackblitz.com/edit/primevue-ts-dt-pt-event)
